server/src/main/java/org/elasticsearch/cluster/coordination/LeaderChecker.java

```diff
@@ -381,17 +381,18 @@ void handleDisconnectedNode(DiscoveryNode discoveryNode) {
 
         private void scheduleNextWakeUp() {
             logger.trace("scheduling next check of {} for [{}] = {}", leader, LEADER_CHECK_INTERVAL_SETTING.getKey(), leaderCheckInterval);
-            transportService.getThreadPool().schedule(new Runnable() {
-                @Override
-                public void run() {
-                    handleWakeUp();
-                }
+            transportService.getThreadPool()
+                .scheduleUnlessShuttingDown(leaderCheckInterval, EsExecutors.DIRECT_EXECUTOR_SERVICE, new Runnable() {
+                    @Override
+                    public void run() {
+                        handleWakeUp();
+                    }
 
-                @Override
-                public String toString() {
-                    return "scheduled check of leader " + leader;
-                }
-            }, leaderCheckInterval, EsExecutors.DIRECT_EXECUTOR_SERVICE);
+                    @Override
+                    public String toString() {
+                        return "scheduled check of leader " + leader;
+                    }
+                });
```

Comment:
We have similar in FollowerChecker, should that be updated as well?

https://github.com/elastic/elasticsearch/blob/d50002853a7d06ac2e07e39169f2a5b5b346673b/server/src/main/java/org/elasticsearch/cluster/coordination/FollowersChecker.java#L404-L416

Reply:
Yeah I don't see why not --> https://github.com/elastic/elasticsearch/pull/108685